store/state/integer.go

```diff
@@ -0,0 +1,48 @@
+package state
+
+type Integer struct {
+	Value
+
+	enc IntEncoding
+}
+
+func NewInteger(v Value, enc IntEncoding) Integer {
+	return Integer{
+		Value: v,
+		enc:   enc,
+	}
+}
+
+func (v Integer) Get(ctx Context) (res uint64) {
```

Comment:
Seems cleaner if we instead internally call `GetSafe` but panic on error.

Reply:
`Get` should panic on unmarshal error, but not on nil value error, so using `GetSafe` internally will cause type switching on errors